### PR TITLE
Update key Id generation logic

### PR DESCRIPTION
**Pull request by sureshchahal**
This code is already deployed in production. We had to fix issues and deploy quickly, so this is change is catching up main repo with current production logic

1. Update keyid logic to generate keys ids between 10 and 99
2. Removed auth from private key endpoint. For alpha, we will reply on attestation report

### CODE REVIEW

File: app.json [hunk 1]
```diff
@@ -104,7 +104,7 @@
         "js_module": "endpoints/keyEndpoint.js",
         "js_function": "unwrapKey",
         "forwarding_required": "sometimes",
-        "authn_policies": ["jwt", "member_cert", "user_cert"],
+        "authn_policies": [],
```
Comment (beejones) [inline app.json:107]: It is better to keep the jwt. The flexibility is in the validation policy where the coordinators can decide which properties are required to provide access to the key endpoints.

File: app.json [hunk 2]
```diff
@@ -171,7 +171,7 @@
         "js_module": "endpoints/keyEndpoint.js",
         "js_function": "key",
         "forwarding_required": "sometimes",
-        "authn_policies": ["jwt", "member_cert", "user_cert"],
+        "authn_policies": [],
```
Comment (beejones) [inline app.json:174]: same here

File: app.json [hunk 3]
```diff
@@ -289,7 +289,7 @@
         "js_module": "endpoints/kms.js",
         "js_function": "hearthbeat",
         "forwarding_required": "sometimes",
-        "authn_policies": ["user_cert", "member_cert", "jwt"],
+        "authn_policies": [],
```
Comment (beejones) [inline app.json:292]: why did you remove this. It provides a nice framework to test authentication

File: src/endpoints/refreshEndpoint.ts [hunk 4]
```diff
@@ -34,7 +34,9 @@ export const refresh = (
     const id = hpkeKeyIdMap.size + 1;
 
     // Generate HPKE key pair with a six digit id
```
Comment (beejones) [inline src/endpoints/refreshEndpoint.ts:36]: the comment is not wrong

File: src/endpoints/refreshEndpoint.ts [hunk 5]
```diff
@@ -34,7 +34,9 @@ export const refresh = (
...
-    const keyItem = KeyGeneration.generateKeyItem(100000 + id);
+    // since OHTTP is limited to 2 char ids, we can only have ids from 10 to 99
+    // So the current logic is to have ids rotate from 10 to 99
+    const keyItem = KeyGeneration.generateKeyItem(id % 90 + 10);
```
Comment (beejones) [inline src/endpoints/refreshEndpoint.ts:39]: I believed that our id was converted to one hex byte in BA services. Is this assumption wrong?

**Comment by beejones** [comment]
The CI did not pass

File: src/endpoints/refreshEndpoint.ts (hunk 4 above)
Comment (beejones) [reply src/endpoints/refreshEndpoint.ts:36]: six digit id?

File: app.json (hunk 1 above)
Comment (sureshchahal) [reply app.json:107]: Didn't we discussed this earlier, and suggestion was to remove it from app.json? I am flexible on approach as long as we don't have any additional process for Ad tech to onboard in alpha.

File: app.json (hunk 3 above)
Comment (sureshchahal) [reply app.json:292]: I wanted an endpoint for appgateway to check whether service is up or not? Since heartbeat endpoint are generally for this purpose, so I remove any auth from it

File: src/endpoints/refreshEndpoint.ts (hunk 4 above)
Comment (sureshchahal) [reply src/endpoints/refreshEndpoint.ts:36]: I will fix this, it should be just 2 digit id

File: src/endpoints/refreshEndpoint.ts (hunk 5 above)
Comment (sureshchahal) [reply src/endpoints/refreshEndpoint.ts:39]: yes, it turned out that BA service need a 2 char string but in as a decimal rather than hex

**Comment by sureshchahal** [comment]
> @sureshchahal The CI fails. Try make demo and npm run test Status: 200 Response data: {"auth":null,"description":["azure-privacy-sandbox-kms","Key Management Service","1.0.0","true"]}
> 
> ✅ [PASS] - Assert OK statusCode
> 
> file:///workspaces/azure-privacy-sandbox-kms/test/e2e-test/src/index.ts:235 const currentValue = toTest[fieldName]; ^
> 
> TypeError: Cannot read properties of null (reading 'policy') at Demo.assertField (file:///workspaces/azure-privacy-sandbox-kms/test/e2e-test/src/index.ts:235:36) at Demo.start (file:///workspaces/azure-privacy-sandbox-kms/test/e2e-test/src/index.ts:69:12) at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
> 
> Node.js v22.9.0
> 
> 💀 Stopped sandbox process due to an error 💥 Error details: Exit Status : 1 Last Command : ${testScript} 2>&1
> 
> ```
> Error Output : No log available
> ```
> 
> make: *** [Makefile:76: demo] Error 1
> 
> Error: Dev container exec failed: (exit code: 2) Error: Dev container exec failed: (exit code: 2)

@beejones  I checked CI result and all e2e tests passed. For npm run test, they pass on my local but look like CI is not able to find them. To me, it seems like a setup issue with CI rather than pull request itself. I am getting different result in local vs server. How can we move forward on this.

I am seeing this in latest CI run

_ ✅ [PASS] - Assert Member 0::key == dPcnQRbpEZXVi+Nj/T5BpHkbEu+Gjct2MFPpR0Etcl8=
  ✅ [PASS] - Assert Member 0::id == 100002
  ✅ [PASS] - Assert headers["cache-control"] == "max-age=254838"
  
  
  ===============================================
  🎉 All Tests Passed...
  ===============================================
  
  
  
  🏁 Test Completed...
  
  💀 Stopped sandbox process
  
  
  > test
  > node --experimental-vm-modules node_modules/jest/bin/jest.js
  
  
  No tests found, exiting with code 1
  Run with `--passWithNoTests` to exit with code 0
  In /workspaces/azure-privacy-sandbox-kms
    143 files checked.
    testMatch: **/test/unit-test/**/*.test.(ts|js|mjs) - 6 matches
    testPathIgnorePatterns: /node_modules/, /lib/ - 134 matches
    testRegex:  - 0 matches
  Pattern:  - 0 matches
  
  Error: Dev container exec failed: (exit code: 1)
  Error: Dev container exec failed: (exit code: 1)
_
  
  ### 
  Local Test run

`Test Suites: 6 passed, 6 total
Tests:       27 passed, 27 total
Snapshots:   0 total
Time:        1.19 s, estimated 3 s
Ran all test suites.`
